Implement sanity checking

yousufAboeldahab157 opened the pull request:
Implemented Sanity Checking on Watchdog

Test F1TENTH FSM (Shows state transitions)
cd ~/ros2_ws/src/watchdog
python3 examples/f1tenth_fsm_example.py

Test Watchdog Node (Shows sensor monitoring)
cd ~/ros2_ws/src/watchdog
source ~/ros2_ws/install/setup.bash
python3 watchdog/watchdog_node.py

Run All Integration Tests
cd ~/ros2_ws/src/watchdog
python3 -m pytest test/test_f1tenth_integration.py -v

Mohammed-Azab committed:
Merge pull request #7 from GIU-F1Tenth/dev

Cleaned up code

yousufAboeldahab157 committed:
feat: Add sanity checking framework foundation

- Add custom ROS2 messages (SanityWarning, SensorHealth, SanitySummary)
- Implement BaseValidator abstract class and SanityChecker orchestrator
- Add comprehensive configuration parameters for sensor validation
- Set up message generation and package dependencies

TODO: Still need sensor validators, watchdog integration, and testing.

yousufAboeldahab157 committed:
feat: implement sensor-specific validators for sanity checking

- Add validators package with __init__.py for module structure
- Implement LiDARValidator with range, noise, and dead zone validation
- Implement BatteryValidator with voltage trends and VESC monitoring
- Implement CameraValidator with quality, brightness, and blur detection
- Implement OdometryValidator with velocity and acceleration limits
- Each validator inherits from BaseValidator with standardized interface
- Add comprehensive validation logic with statistical analysis

Still needed: cross-sensor validation, watchdog node integration, package dependencies update, testing suite, and documentation

yousufAboeldahab157 committed:
feat: Add F1TENTH integration framework and clean documentation

- Add F1TENTH-specific launch files and racing-optimized configuration
- Implement FSM integration with health-based state transitions
- Add graceful import handling for ackermann_msgs dependencies
- Create comprehensive integration tests (5/5 passing)
- Standardize documentation to professional production standards
- Enhanced SanityChecker with watchdog compatibility methods

Ready for F1TENTH autonomous racing deployment.

yousufAboeldahab157 committed:
feat: Add F1TENTH integration framework and clean documentation

- Add F1TENTH-specific launch files and racing-optimized configuration
- Implement FSM integration with health-based state transitions
- Add graceful import handling for ackermann_msgs dependencies
- Create comprehensive integration tests (5/5 passing)
- Standardize documentation to professional production standards
- Enhanced SanityChecker with watchdog compatibility methods

Ready for F1TENTH autonomous racing deployment.

Mohammed-Azab commented on launch/f1tenth_system.launch.py:
why there is 2 diff launch files? One is sufficient

Mohammed-Azab commented on examples/f1tenth_fsm_example.py:
is it just for illustrations? if so can u remove it since we currently don't have a stable version of FSM

yousufAboeldahab157 replied:
yes this was mainly a placeholder file for a test, i forgot to remove it

Mohammed-Azab commented on install/.colcon_install_layout:
Install Folder should be in gitignore, rebase the last commit with it.

Mohammed-Azab commented on log/COLCON_IGNORE:
Log Folder should be in gitignore, rebase the last commit with it.

yousufAboeldahab157 replied:
done

Mohammed-Azab commented on msg/SanitySummary.msg:
why there is more than 1 custom msg, can you write what is the diff ? or can you integrate them into single file ? 
Also adding custom msgs is only allowed in
https://github.com/GIU-F1Tenth/giu_f1t_interfaces

Mohammed-Azab commented on package.xml:
changing from ament_python to ament_cmake will make the pkg build for custom msg only, and the watchdog pkg will not build successfully, that's why we have a seperate repo for the custom msgs. 
[giu_f1t_interfaces](https://github.com/GIU-F1Tenth/giu_f1t_interfaces)

yousufAboeldahab157 replied:
fixed

Mohammed-Azab commented on docs/F1TENTH_INTEGRATION.md:
```diff
@@ -0,0 +1,347 @@
+- Optional packages:
+  - `robot_localization` (EKF)
+  - `realsense2_camera` (Intel RealSense)
+  - `urg_node` (Hokuyo LiDAR)
```
how is this an optional pkg, since it's the most critical component in the car.

yousufAboeldahab157 replied:
fixed

Mohammed-Azab commented on config/f1tenth_params.yaml:
can you have only 1 config file ?

Mohammed-Azab commented on F1TENTH_STATUS.md:
what is F1tenth Status ? is an extened version of README

Mohammed-Azab commented on watchdog/watchdog_main.py:
why this file exist since we have the actual node ?

yousufAboeldahab157 replied:
fixed

Mohammed-Azab commented on test/run_tests.py:
if it's needed pls remove it

Mohammed-Azab commented on setup.py:
```diff
@@ -25,6 +30,7 @@
     entry_points={
         'console_scripts': [
             'watchdog_node = watchdog.watchdog_node:main',
+            'fsm_integration_node = watchdog.fsm_integration:main',
```
why a seperate node ?

yousufAboeldahab157 replied:
setup.py is needed for ament_python package

Mohammed-Azab commented on watchdog/fsm_integration.py:
HOW DOES THIS EXIST AND WE DON'T HAVE FSM?

yousufAboeldahab157 replied:
placeholder for a test now removed

Mohammed-Azab reviewed (changes requested):
Please review my comments and rebase ur branch so that the log, install, build files can't be uploaded to github.  Also I don't really understand what is FSM integration, Watchdog system should be independent from any behaviour controller.

FamALouiz commented:
@yousufAboeldahab157 small ping :)

Also, please merge to dev before main. Thank you!

yousufAboeldahab157 committed:
chore: stop tracking log/ (use .gitignore)

yousufAboeldahab157 committed:
changes according to pull request

Mohammed-Azab commented on config/watchdog_params.yaml:
still why we have 2 diff launch files ?

yousufAboeldahab157 replied:
deleted the other one

Mohammed-Azab commented on install/local_setup.bash:
you added them in the gitignore but they still exist here. you should have rebased ur branch not making another commit. 
So pls rebase ur branch or do anything else to drop these files from getting to the dev branch

yousufAboeldahab157 committed:
chore: address PR unresolved threads

- Stop tracking install/ (already in .gitignore) to ensure build artifacts never enter the repo
- Remove deprecated FSM integration module (watchdog/fsm_integration.py)
- Remove duplicate F1TENTH launch files; keep single canonical launch/watchdog.launch.py

This resolves:
- comment on FSM integration existence
- comment about having 2 different launch files
- comment about install/ files appearing in PR

yousufAboeldahab157 committed:
fix: remove unnecessary launch file
